lib/lacework.bash

```diff
@@ -0,0 +1,276 @@
+            if [ -z "${PROFILE}" ]; then
+                echo "ERROR: Missing config related to IAC scans. Need the following: IAC_SCAN_TYPE, LW_ACCOUNT, LW_API_KEY, LW_API_SECRET" >&2
+                exit 1
+            fi
+        fi
+    fi
+
+    #vuln scan related env variables
+    if [ "${SCAN_TYPE}" == "vulnerability" ]; then
+        export ACCESS_TOKEN_ENV_VAR="${BUILDKITE_PLUGIN_LACEWORK_ACCESS_TOKEN_ENV_VAR:-}"
+        export VULNERABILITY_SCAN_REPOSITORY="${BUILDKITE_PLUGIN_LACEWORK_VULNERABILITY_SCAN_REPOSITORY:-}"
+        export VULNERABILITY_SCAN_TAG="${BUILDKITE_PLUGIN_LACEWORK_VULNERABILITY_SCAN_TAG:-}"
+        if [ -z "${ACCESS_TOKEN_ENV_VAR}" ] || [ -z "${VULNERABILITY_SCAN_REPOSITORY}" ] || [ -z "${VULNERABILITY_SCAN_TAG}" ]; then
+            echo "ERROR: Missing config related to vulnerability scans. Need the following: ACCESS_TOKEN_ENV_VAR, VULNERABILITY_SCAN_REPOSITORY, VULNERABILITY_SCAN_TAG" >&2
+            exit 1
+        fi
+    fi
+
+    export FAIL_LEVEL="${BUILDKITE_PLUGIN_LACEWORK_FAIL_LEVEL:-}"
+
+}
+
+# Lacework Scans based on the provided scan tool
+function lacework_scan() {
+    case "${SCAN_TYPE}" in
+        sca)
+        lacework_sca
+        ;;
+
+        iac)
+        lacework_iac
+        ;;
+
+        vulnerability)
+        lacework_vulnerability
+        ;;
+
+        sast)
+        lacework_sast
+        ;;
+    esac
+}
+
+
+function lacework_sca() {
+
+    echo "--- Running Lacework SCA scan"
+
+    CMD=(
+    lacework
+    )
+
+    if [ -n "$PROFILE" ]; then
+        CMD+=(
+            --profile "$PROFILE"
+        )
+    else
+        CMD+=(
+        --account "${ACCOUNT_NAME}"
+        --api_key "${API_KEY_ENV_VAR}"
+        --api_secret "${API_KEY_SECRET_ENV_VAR}"
+    )
+    fi  
+
+    CMD+=(
+        sca scan .
+        --save-results
+    )
+        
+
+    echo "${CMD[@]}"
+
+    "${CMD[@]}" 
+}
+
+function lacework_sast() {
+
+    echo "--- Running Lacework SAST scan"
+
+    CMD=(
+    lacework
+    )
+
+    if [ -n "$PROFILE" ]; then
+        CMD+=(
+            --profile "$PROFILE"
+        )
+    else
+        CMD+=(
+        --account "${ACCOUNT_NAME}"
+        --api_key "${API_KEY_ENV_VAR}"
+        --api_secret "${API_KEY_SECRET_ENV_VAR}"
+        )
+    fi  
+
+    
+
+    SARIF_ARTIFACT="lacework-sast-report-${BUILDKITE_PIPELINE_SLUG}-${BUILDKITE_BUILD_NUMBER}.sarif"
+
+    CMD+=(
+        sast scan -o "${SARIF_ARTIFACT}"
+    )
+        
+
+    echo "${CMD[@]}"
+
+    "${CMD[@]}" 
+
+    annotate_and_upload_build_sast "${SARIF_ARTIFACT}"
+}
+
+function lacework_iac() {
+
+    echo "--- Running Lacework IAC scan"
+
+    CMD=(
+    lacework
+    )
+
+
+    CMD+=(
+        iac
+        "${IAC_SCAN_TYPE}"
+    )
+
+    if [ -n "$PROFILE" ]; then
+    CMD+=(
+        --iac-profile "$PROFILE"
+    )
+    fi  
+
+    if [ -n "$FAIL_LEVEL" ]; then
+    CMD+=(
+        --fail "$FAIL_LEVEL"
+    )
+    fi  
+
+
+    echo "${CMD[@]}"
```

Comment:
```suggestion

```
Although no Secrets are exposed for this function, would be good to change this for continunity with other functions